File: app/src/main/java/com/ladsoft/bakingapp/data/database/translator/RecipeRecordTranslator.java
package com.ladsoft.bakingapp.data.database.translator;

import com.ladsoft.bakingapp.data.database.entity.RecipeRecord;
import com.ladsoft.bakingapp.entity.Recipe;
import com.ladsoft.bakingapp.translator.Translator;

public class RecipeRecordTranslator implements Translator<Recipe, RecipeRecord> {
    @Override
    public RecipeRecord translate(Recipe source) {
        return new RecipeRecord(source.getId(),
                        source.getName() != null ? source.getName() : "",
                        source.getServings(),
                        source.getImageUrl() != null ? source.getImageUrl() : "");
    }
}

File: app/src/main/java/com/ladsoft/bakingapp/data/database/translator/RecipeRecordTranslator.kt
package com.ladsoft.bakingapp.data.database.translator

import com.ladsoft.bakingapp.data.database.entity.RecipeRecord
import com.ladsoft.bakingapp.entity.Recipe
import com.ladsoft.bakingapp.translator.Translator

class RecipeRecordTranslator: Translator<Recipe, RecipeRecord> {
    override fun translate(source: Recipe): RecipeRecord {
        return RecipeRecord(source.id, source.name, source.servings, source.imageUrl)
    }
}